app/src/main/java/com/example/testandroidapplication/View/createProfile/VenueProfileCreation.java

```diff
@@ -0,0 +1,181 @@
+import com.example.testandroidapplication.R;
+import com.example.testandroidapplication.utils.CheckNetworkStatus;
+import com.example.testandroidapplication.objects.User;
+import com.example.testandroidapplication.objects.Venue;
+
+public class VenueProfileCreation extends Fragment implements IVenueProfileCreationContract.View{
+
+
+    private User user;
+    private User.UserBuilder userBuilder;
+    private VenueProfileCreationPresenter presenter;
+
+    private EditText venueNameEditText;
+    private EditText venueTaglineEditText;
+    private EditText venueLocationEditText;
+    private EditText venueDescriptionEditText;
+    private EditText venueFacebookEditText;
+    private EditText venueTwitterEditText;
+    private EditText venueInstagramEditText;
+    private EditText venueWebsiteEditText;
+    private EditText venueAddress1EditText;
+    private EditText venuePostcodeEditText;
+
+    private String venueNameInput;
+    private String venueTaglineInput;
+    private String venueLocationInput;
+    private String venueDescriptionInput;
+    private String venueFacebookInput;
+    private String venueTwitterInput;
+    private String venueInstagramInput;
+    private String venueWebsiteInput;
+    private String venueAddress1Input;
+    private String venuePostcodeInput;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+
+        final View v = inflater.inflate(R.layout.venue_profile_creation, container, false);
+
+        presenter = new VenueProfileCreationPresenter(this);
+
+        venueNameEditText = v.findViewById(R.id.venue_name);
+        venueTaglineEditText = v.findViewById(R.id.venue_tagline);
+        venueLocationEditText = v.findViewById(R.id.venue_location);
+        venueDescriptionEditText = v.findViewById(R.id.venue_description);
+        venueFacebookEditText = v.findViewById(R.id.venue_facebook);
+        venueTwitterEditText = v.findViewById(R.id.venue_twitter);
+        venueInstagramEditText = v.findViewById(R.id.venue_instagram);
+        venueWebsiteEditText = v.findViewById(R.id.venue_webpage);
+        venueAddress1EditText = v.findViewById(R.id.venue_address_line_1);
+        venuePostcodeEditText = v.findViewById(R.id.venue_address_line_postcode);
+
+        Button createProfileBtn = v.findViewById(R.id.venue_create_profile);
+
+        createProfileBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // if network is available, validate the user input
+                if (CheckNetworkStatus.isNetworkAvailable(getActivity().getApplicationContext())) {
+
+                    userBuilder = new User.UserBuilder("1", "Stone Temple Pilots", "stone@temple.com");
+                    user = userBuilder.build();
+
+                    venueNameInput = venueNameEditText.getText().toString();
```

Comment:
These repeated methods will eventually replaced by an iterator that runs through a mapping class. Validation will be delegated to the Presenter, which will know the business logic for each input from the respective Data models